lib/sources.nix

```diff
@@ -254,6 +254,64 @@ let
       outPath = builtins.path { inherit filter name; path = origSrc; };
     };
 
+  # urlToName : (URL | Path | String) -> String
+  #
+  # Transform a URL (or path, or string) into a clean package name.
```

Comment:
Maybe I can be wrong, but the general term is URI.

Reply:
As a thing more general than a URL? Yes. But here it is literally "a URL, or a path, or a string (a package name, usually)".